packages/babel-preset/src/transformPlugin.ts

```diff
@@ -62,6 +77,90 @@ function getCalleeFunctionKind(
   return null;
 }
 
+/**
+ * Gets the id of the parent variable declarator
+ */
+function getParentDeclaratorId(path: NodePath<t.CallExpression> | NodePath<t.MemberExpression>): string {
+  const declarator = path.findParent(p => p.isVariableDeclarator());
+  if (declarator?.isVariableDeclarator()) {
+    const id = declarator.get('id');
+    if (id.isIdentifier()) {
+      return id.node.name;
+    }
+  }
+
+  return 'unknownHook';
+}
+
+/**
+ * Extracts CSS rules from evaluated reset styles to metadata
+ */
+function buildCSSResetEntriesMetadata(
+  state: BabelPluginState,
+  cssRules: string[] | CSSRulesByBucket,
+  hookName: string,
+) {
+  state.cssResetEntries ??= {};
+  state.cssResetEntries[hookName] ??= [];
+  if (Array.isArray(cssRules)) {
+    state.cssResetEntries[hookName] = cssRules;
+  } else {
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    for (const [_, bucketEntries] of Object.entries(cssRules)) {
+      for (const bucketEntry of bucketEntries) {
+        if (Array.isArray(bucketEntry)) {
+          state.cssResetEntries[hookName].push(bucketEntry[0]);
+        } else {
+          state.cssResetEntries[hookName].push(bucketEntry);
+        }
+      }
+    }
+  }
+}
+
+/**
+ * Extracts CSS rules from evaluated styles to metadata
+ */
+function buildCSSEntriesMetadata(
+  state: BabelPluginState,
+  classnamesMapping: CSSClassesMapBySlot<string>,
+  cssRulesByBucket: CSSRulesByBucket,
+  hookName: string,
+) {
+  const classesBySlot: Record<string, string[]> = {};
+  for (const [slot, cssClassesMap] of Object.entries(classnamesMapping)) {
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    for (const [_, cssClasses] of Object.entries(cssClassesMap)) {
+      classesBySlot[slot] = Array.isArray(cssClasses) ? cssClasses : [cssClasses];
+    }
+  }
+
+  const cssRules: string[] = [];
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  for (const [_, bucketEntries] of Object.entries(cssRulesByBucket)) {
+    for (const bucketEntry of bucketEntries) {
+      if (Array.isArray(bucketEntry)) {
+        cssRules.push(bucketEntry[0]);
+      } else {
+        cssRules.push(bucketEntry);
+      }
+    }
+  }
+
+  state.cssEntries ??= {};
+  state.cssEntries[hookName] ??= {};
+
+  for (const [slot, cssClasses] of Object.entries(classesBySlot)) {
+    for (const cssClass of cssClasses) {
+      const cssRule = cssRules.find(rule => rule.includes(cssClass));
+      if (cssRule) {
+        state.cssEntries[hookName][slot] ??= [];
+        state.cssEntries[hookName][slot].push(cssRule);
+      }
+    }
+  }
+}
```

Comment:
```suggestion
function buildCSSEntriesMetadata(
  state: Required<BabelPluginState>,
  classnamesMapping: CSSClassesMapBySlot<string>,
  cssRulesByBucket: CSSRulesByBucket,
  hookName: string,
) {
  const classesBySlot: Record<string, string[]> = Object.fromEntries(
    Object.entries(classnamesMapping).map(([slot, cssClassesMap]) => {
      return [
        slot,
        Object.values(cssClassesMap).flatMap(cssClasses => (Array.isArray(cssClasses) ? cssClasses : [cssClasses])),
      ];
    }),
  );

  const cssRules: string[] = Object.values(cssRulesByBucket).flatMap(cssRulesByBucket => {
    return cssRulesByBucket.map(bucketEntry => {
      const [cssRule] = normalizeCSSBucketEntry(bucketEntry);

      return cssRule;
    });
  });

  state.cssEntries[hookName] = Object.fromEntries(
    Object.entries(classesBySlot).map(([slot, cssClasses]) => {
      return [
        slot,
        cssClasses.map(cssClass => {
          return cssRules.find(rule => rule.includes(cssClass))!;
        }),
      ];
    }),
  );
}
```

This also fixes broken part